pallets/utxo/src/staking.rs

```diff
@@ -0,0 +1,187 @@
+
+    /// transfer balance from the locked state to the actual free balance.
+    fn withdraw(controller_account: &AccountId) -> DispatchResultWithPostInfo;
+
+}
+
+/// performs the staking OUTSIDE of the `pallet-utxo`. Calls the `fn stake(...)` function of
+/// `StakingHelper` trait.
+/// # Arguments
+/// * `stash_pubkey` - is a `validator` in terms of `pallet-staking`; but in the utxo, this is nothing.
+/// This is only to satisfy the `pallet-staking`. BUT this acts as a "bank", so this account MUST have
+/// funds in its `pallet-balances` counterpart.
+/// * `controller_pubkey` - is the `validator` in our utxo system.
+/// * `session_key` - to get up-to-date with the validators, eras, sessions. see `pallet-session`.
+/// * `value` - the amount to stake/bond/stash
+pub(crate) fn stake<T: Config>(
+    stash_pubkey: &H256,
+    controller_pubkey: &H256,
+    session_key:&Vec<u8>,
+    value:Value
+)
+    -> DispatchResultWithPostInfo {
+    ensure!(!<StakingCount<T>>::contains_key(controller_pubkey),Error::<T>::StakingAlreadyExists);
+
+    let stash_account = T::StakingHelper::get_account_id(stash_pubkey);
+    let controller_account = T::StakingHelper::get_account_id(controller_pubkey);
+
+    T::StakingHelper::stake(&stash_account, &controller_account, session_key,value)
+}
+
+/// stake more values. This is only for existing validators.
+pub(crate) fn stake_extra<T: Config>(controller_pubkey: &H256, value:Value) -> DispatchResultWithPostInfo {
+    // Checks whether a given pubkey is a validator
+    ensure!(<StakingCount<T>>::contains_key(controller_pubkey.clone()) , Error::<T>::NoStakingRecordFound);
+
+    let controller_account= T::StakingHelper::get_account_id(controller_pubkey);
+
+    T::StakingHelper::stake_extra(&controller_account,value)
+}
+
+/// unlocking the staked funds outside of the `pallet-utxo`.
+/// also means quitting/pausing from being a validator.
+pub(crate) fn unlock<T: Config>(controller_pubkey: &H256) -> DispatchResultWithPostInfo {
+    let controller_account= T::StakingHelper::get_account_id(controller_pubkey);
+    T::StakingHelper::pause(&controller_account)?;
+
+    <Pallet<T>>::deposit_event(Event::<T>::StakeUnlocked(controller_pubkey.clone()));
+
+    Ok(().into())
+}
+
+/// Consolidates all unlocked utxos  into one, and moves it to `UtxoStore`.
+/// Make SURE that `fn unlock(...)` has been called and the era for withdrawal has passed, before
+/// performing a withdrawal.
+/// # Arguments
+/// * `controller_pubkey` - the public key of the validator. In terms of pallet-staking, that's the
+/// controller, NOT the stash.
+/// * `outpoints` - a list of outpoints that were staked.
+pub(crate) fn withdraw<T: Config>(controller_pubkey: H256, outpoints: Vec<H256>) -> DispatchResultWithPostInfo {
+    validate_withdrawal::<T>(&controller_pubkey,&outpoints)?;
+
+    let controller_account = T::StakingHelper::get_account_id(&controller_pubkey);
+    T::StakingHelper::withdraw(&controller_account)?;
```

Comment:
Please ensure the authentication of this process and make a test that attempts to withdraw without having authority to do so.